programs/application/src/lib.rs

```diff
@@ -19,9 +19,9 @@ pub mod application {
 }
 
 #[derive(Accounts)]
-#[instruction(job_ad_id: String)]
+#[instruction(job_ad_id: String, application_id: String)]
 pub struct Initialize<'info> {
-    #[account(init, payer = authority, seeds = [APPLICATION_SEED, job_ad_id.as_bytes()[..18].as_ref(), job_ad_id.as_bytes()[18..].as_ref(), authority.key().as_ref()], bump , space = 4 + 32 + 1 + 8 )]
+    #[account(init, payer = authority, seeds = [APPLICATION_SEED, job_ad_id.as_bytes()[..18].as_ref(), job_ad_id.as_bytes()[18..].as_ref(), application_id.as_ref()], bump , space = 4 + 32 + 1 + 8 )]
```

Comment:
application id will also be an `uuid`. So we need to do the trick of splitting the bytes.
I don't think we need to store the `job_ad` in the seed. If you need to retrieve that value, then it can be stored in a PDA for instance. But I don't think we get anything by adding it as a seed, since the job application id is unique.